app/api/chat/engine/chat.ts

```diff
@@ -1,30 +1,68 @@
-import { ContextChatEngine, Settings, SimpleChatEngine } from "llamaindex";
+import {
+  BaseToolWithCall,
+  MetadataFilter,
+  MetadataFilters,
+  OpenAIAgent,
+  QueryEngineTool,
+  Settings,
+  SimpleChatEngine,
+} from "llamaindex";
 import { getDataSource } from "./index";
-import { STORAGE_CACHE_DIR } from "./shared";
 
 interface ChatEngineOptions {
   datasource?: string;
+  documentIds?: string[];
 }
 
-export async function createChatEngine({ datasource }: ChatEngineOptions) {
-  if (datasource) {
-    const index = await getDataSource(datasource);
-    if (!index) {
-      throw new Error(
-        `No datasources found in storage cache folder: ${STORAGE_CACHE_DIR}/${datasource}. Run generate it first.`,
-      );
-    }
-    const retriever = index.asRetriever({
-      similarityTopK: process.env.TOP_K ? parseInt(process.env.TOP_K) : 3,
-    });
-    return new ContextChatEngine({
-      chatModel: Settings.llm,
-      retriever,
-      systemPrompt: process.env.SYSTEM_PROMPT,
+export async function createChatEngine({
+  datasource,
+  documentIds,
+}: ChatEngineOptions) {
+  if (!datasource) {
+    return new SimpleChatEngine({
+      llm: Settings.llm,
     });
   }
 
-  return new SimpleChatEngine({
-    llm: Settings.llm,
+  const index = await getDataSource(datasource);
```

Comment:
TODO: use ContextChatEngine after https://github.com/run-llama/create-llama/pull/196/files is released